storage/engine/engine_test.go

```diff
@@ -663,20 +663,38 @@ func TestSnapshotMethods(t *testing.T) {
 			t.Error("expected invalid iterator when seeking to element which shouldn't be visible to snapshot")
 		}
 
-		// Verify NewSnapshot returns nil.
-		if newSnap := snap.NewSnapshot(); newSnap != nil {
-			t.Error("expected NewSnapshot on snapshot to return nil; got %+v", newSnap)
+		// Verify Commit is error.
+		if err := snap.Commit(); err == nil {
+			t.Error("expected error on Commit to snapshot")
 		}
+	}, t)
+}
 
-		// Verify NewBatch returns nil.
-		if batch := snap.NewBatch(); batch != nil {
-			t.Error("expected NewBatch on snapshot to return nil; got %+v", batch)
+// TestSnapshotNewSnapshot panics.
+func TestSnapshotNewSnapshot(t *testing.T) {
+	defer func() {
```

Comment:
Put the defer inside the function at line 681 so it gets run for each engine; I think when the defer is here you're only testing the first engine.